bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/provisional/p2/repository/DefaultPGPPublicKeyService.java

```diff
@@ -742,9 +742,12 @@ private static abstract class GPGPubringSupplier implements Supplier<PGPPublicKe
 
 		public GPGPubringSupplier(Path pubring) {
 			this.pubring = pubring;
-			
+			try {
 				keyRingCollection = new PGPPublicKeyRingCollection(Collections.emptyList());
-			
+			} catch (Exception e) {
+				// Cannot happen for an empty collection.
+				throw new RuntimeException(e);
```

Comment:
If it can not happen throw Assertion Error with explanation message, or if we just think it *should* not happen one should first check if `e` is a Runtime Exception already before wrapping it again.

Reply:
Would this approach be okay? 
catch (Exception e) {
    if (e instanceof RuntimeException) {
        throw (RuntimeException) e;
    } else {
        throw new RuntimeException("Wrapping exception", e); 
    }

Reply:
A message like `throw new RuntimeException("Unexpected exception while PGPPublicKeyRingCollection", e);` would be better but basically yes.